letsencrypt_auto/letsencrypt-auto.template

```diff
@@ -141,65 +81,146 @@ then
   elif grep -iq "Amazon Linux" /etc/issue ; then
-    ExperimentalBootstrap "Amazon Linux" _rpm_common.sh
+    ExperimentalBootstrap "Amazon Linux" BootstrapRpmCommon
   else
     echo "Sorry, I don't know how to bootstrap Let's Encrypt on your operating system!"
     echo
-    echo "You will need to bootstrap, configure virtualenv, and run a pip install manually"
+    echo "You will need to bootstrap, configure virtualenv, and run a peep install manually."
     echo "Please see https://letsencrypt.readthedocs.org/en/latest/contributing.html#prerequisites"
-    echo "for more info"
+    echo "for more info."
   fi
+}
 
-  DeterminePythonVersion
-  echo "Creating virtual environment..."
-  if [ "$VERBOSE" = 1 ] ; then
-    virtualenv --no-site-packages --python $LE_PYTHON $VENV_PATH
+# This script takes the same arguments as the main letsencrypt program, but it
+# additionally responds to --verbose (more output) and --debug (allow support
+# for experimental platforms)
+for arg in "$@" ; do
+  # This first clause is redundant with the third, but hedging on portability
+  if [ "$arg" = "-v" ] || [ "$arg" = "--verbose" ] || echo "$arg" | grep -E -- "-v+$" ; then
+    VERBOSE=1
+  elif [ "$arg" = "--debug" ] ; then
+    DEBUG=1
+  fi
+done
+
+# letsencrypt-auto needs root access to bootstrap OS dependencies, and
+# letsencrypt itself needs root access for almost all modes of operation
+# The "normal" case is that sudo is used for the steps that need root, but
+# this script *can* be run as root (not recommended), or fall back to using
+# `su`
+if test "`id -u`" -ne "0" ; then
+  if command -v sudo 1>/dev/null 2>&1; then
+    SUDO=sudo
   else
-    virtualenv --no-site-packages --python $LE_PYTHON $VENV_PATH > /dev/null
+    echo \"sudo\" is not available, will use \"su\" for installation steps...
+    # Because the parameters in `su -c` has to be a string,
+    # we need properly escape it
+    su_sudo() {
+      args=""
+      # This `while` loop iterates over all parameters given to this function.
+      # For each parameter, all `'` will be replace by `'"'"'`, and the escaped string
+      # will be wrap in a pair of `'`, then append to `$args` string
+      # For example, `echo "It's only 1\$\!"` will be escaped to:
+      #   'echo' 'It'"'"'s only 1$!'
+      #     │       │└┼┘│
+      #     │       │ │ └── `'s only 1$!'` the literal string
+      #     │       │ └── `\"'\"` is a single quote (as a string)
+      #     │       └── `'It'`, to be concatenated with the strings followed it
+      #     └── `echo` wrapped in a pair of `'`, it's totally fine for the shell command itself
+      while [ $# -ne 0 ]; do
+        args="$args'$(printf "%s" "$1" | sed -e "s/'/'\"'\"'/g")' "
+        shift
+      done
+      su root -c "$args"
+    }
+    SUDO=su_sudo
   fi
 else
-  DeterminePythonVersion
+  SUDO=
 fi
 
+if [ "$1" = "--os-packages-only" ]; then
+  Bootstrap
+elif [ "$1" != "--_skip-to-install" ]; then
```

Comment:
No; if someone manually passes this flag, it could cause le-auto to try installing Python packages without a virtualenv made.